src/http-api/utils.ts

```diff
@@ -79,17 +79,29 @@ export function parseErrorResponse(response: XMLHttpRequest | Response, body?: s
         return <Error>e;
     }
 
-    if (contentType?.type === "application/json" && body) {
-        return new MatrixError(
-            JSON.parse(body),
-            response.status,
-            isXhr(response) ? response.responseURL : response.url,
-        );
-    }
-    if (contentType?.type === "text/plain") {
-        return new HTTPError(`Server returned ${response.status} error: ${body}`, response.status);
-    }
-    return new HTTPError(`Server returned ${response.status} error`, response.status);
+    const httpHeaders = !isXhr(response)
+        ? response.headers
+        : new Headers(
```

Comment:
this would probably be clearer the other way around?

```
    const httpHeaders = isXhr(response)
        ? new Headers( ... )
        : response.headers;
```